File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
    
    result := make([]int,0)
    
    var postOrder func(*TreeNode)
    
    postOrder = func(node *TreeNode) {
        if node == nil {
            return
        }
        
        postOrder(node.Left)
        postOrder(node.Right)
        result = append(result, node.Val)
    }
    
    postOrder(root)
    
    return result
}   

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> list = new ArrayList();
        postOrderTraversal(root, list);
        return list;
    }
    
    private void postOrderTraversal(TreeNode root, List<Integer> list) {
        if (root == null) {
            return;
        }
        
        postOrderTraversal(root.left, list);
        postOrderTraversal(root.right, list);
        list.add(root.val);
    }
}